src/extensionInstall.mts

```diff
@@ -0,0 +1,71 @@
+import electron, { ProgressReporter } from '@vscode/test-electron';
+import { spawnSync } from 'node:child_process';
+import { createRequire } from 'node:module';
+import path from 'node:path';
+
+const require = createRequire(import.meta.url);
+
+export async function installExtensions(
+  extensionDevelopmentPath: string | string[],
+  extensions: string[] = [],
+  installDependentExtensions: boolean = false,
+  codeVersion?: string,
+  desktopPlatform?: string,
+  reporter?: ProgressReporter,
+) {
+  const vscodePath = await electron.downloadAndUnzipVSCode(codeVersion, desktopPlatform, reporter);
+
+  const extensionDevelopmentPaths = Array.isArray(extensionDevelopmentPath)
+    ? extensionDevelopmentPath
+    : [extensionDevelopmentPath];
+
+  const extensionsToInstall = installDependentExtensions
+    ? mergeDependentExtensions(extensions, extensionDevelopmentPaths)
+    : extensions;
+
+  const [cli, ...cliArgs] = electron.resolveCliArgsFromVSCodeExecutablePath(vscodePath, {
+    platform: desktopPlatform,
+  });
+
+  for (const extension of extensionsToInstall) {
+    cliArgs.push('--install-extension', extension);
+  }
+
+  console.debug('Installing VSCode Extensions', cli, cliArgs);
+
+  // TODO: Start async and stream the results
+  const installResult = spawnSync(cli, cliArgs, { encoding: 'utf-8' });
+  if (installResult.stderr) {
+    throw new Error(`Error installing extensions: ${installResult.stderr}`);
+  }
+  return installResult.stdout;
+}
+
+// Merge in the dependent extensions from the package.json file only if the base name prior to the @ symbol is not already specified in extensions
+function mergeDependentExtensions(extensions: string[], extensionDevelopmentPaths: string[]) {
+  const extensionsToInstall = new Set(extensions);
```

Comment:
Using a set here isn't giving us much, we transform it to an array every time we look at it

Reply:
The main thing was to keep things deduplicated to avoid duplicate entries but there's better ways I can handle that, agreed.

Reply:
0149caca3cfd38148499b5a782f3c191e95b50a8